src/providers/google-reverse-geocoder.php

```diff
@@ -2,11 +2,30 @@
 
 namespace SearchApi\Providers;
 
-use SearchApi\Services\ReverseGeocoder;
+use SearchApi\Models as Models;
+use SearchApi\Support as Support;
+use SearchApi\Test\Support\Mocks as Mocks;
+use SearchApi\Services\ReverseGeocoder as ReverseGeocoder;
 
 /**
- * Class GoogleReverseGeocoder - Given a location name find a point or polygon that can represent that
+ * Class GoogleReverseGeocoder - Given a latlin or place id to find a point or polygon that can represent that
  * location via GoogleMaps.
  */
 class GoogleReverseGeocoder implements ReverseGeocoder {
+  public $api_key;
+
+  public function __construct( $key = null ) {
+  	$this->api_key = $key;
+  }
+
+  public function get_locations( Models\GeoCoordinate $geo_coordinate ) {
+  	$parser = new Support\GeoParser();
+  	$mock = new Mocks\Google_Geocoder_Mock();
+  	$geocoding_results = $mock->reverse_geocoding( $geo_coordinate, $this->api_key );
+  	if ( $geocoding_results === 'Invalid Key' ) {
+  	  return 'Invalid Key';
```

Comment:
Returning a string as an error state might not be the easiest thing for who ever is using this to catch. Especially when it looks like your actual data being returned is also be a string.